### PR TITLE
Add rustup.

### PR DESCRIPTION
**Pull request by dlorenc**
This is needed for our rust image.

Signed-off-by: Dan Lorenc <dlorenc@chainguard.dev>

### CODE REVIEW

**Comment by dlorenc** [comment]
Ref https://github.com/chainguard-images/images/issues/246

File: rustup.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      - busybox
+      - make
+      - git
+      - ca-certificates-bundle
+      - build-base
+      - openssl-dev
+      - python3
+
+pipeline:
+  # There's some autodetectin in the buid that requires it to run from a git repository.
```
Comment (amouat) [inline rustup.yaml:29]: pedantic nit: "autodetectin" and "buid"

File: rustup.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package:
+  name: rustup
+  version: 1.25.1
+  epoch: 0
+  description: "Spin is an open source framework for building and running fast, secure, and composable cloud microservices with WebAssembly"
```
Comment (amouat) [inline rustup.yaml:5]: Guess this needs updated :)

**Review by amouat — CHANGES_REQUESTED** [review]
Description needs updated but looks good.

File: rustup.yaml (hunk 2 above)
Comment (dlorenc) [reply rustup.yaml]: Fixed!

File: rustup.yaml (hunk 1 above)
Comment (dlorenc) [reply rustup.yaml]: Fixed!

**Review by rawlingsj — DISMISSED** [review]

File: rustup.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package:
+  name: rustup
+  version: 1.25.1
+  epoch: 0
+  description: "rustup is an installer for
+the systems programming language Rust"
```
Comment (amouat) [inline rustup.yaml:6]: This seems to have worked, but I think I'd rather get rid of the new line.

**Review by amouat — DISMISSED** [review]
Maybe delete the newline in description but looks good.

File: rustup.yaml (hunk 3 above)
Comment (dlorenc) [reply rustup.yaml]: Done!

**Review by rawlingsj — APPROVED** [review]

**Review by jdolitsky — APPROVED** [review]

**Pull request merged**